src/parser/core/Combatant.ts

```diff
@@ -105,13 +96,16 @@ class Combatant extends Entity {
     };
 
     this._parseTalents(combatantInfo.talents);
-    this._parseTraits(combatantInfo.artifact);
-    this._parseEssences(combatantInfo.heartOfAzeroth);
-    this._parseCovenant(combatantInfo.covenant);
-    this._parseSoulbind(combatantInfo.soulbind);
-    this._parseConduits(combatantInfo.conduits);
+    this._parseTraits(combatantInfo.artifact as Trait[]);
+    this._parseEssences(combatantInfo.heartOfAzeroth as Essence[]);
     this._parseGear(combatantInfo.gear);
     this._parsePrepullBuffs(combatantInfo.auras);
+    if (combatantInfo.expansion === 'shadowlands') {
+      this._parseCovenant(combatantInfo.covenantID);
+      this._parseSoulbind(combatantInfo.soulbindID);
+      this._parseSoulbindTraits(combatantInfo.artifact as SoulbindTrait[]);
```

Comment:
its not a huge deal to resolve these, but the way to do it without casting is typeguards  https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
Interesting idea, which I think we should probably do if they keep the naming scheme once SL goes live and it's not just some temporary beta-naming-scheme, because it just doesn't make sense to have them be called artifact and heartOfAzeroth 😄